src/plugins/event-plugins/__tests__/WebVitalsPlugin.test.ts

```diff
@@ -47,11 +52,41 @@ const mockCLSData = {
     }
 };
 
+// only need hasLatency fields
+const imagePerformanceEntry = {
+    duration: 50,
+    startTime: 100
+};
+
+const imageResourceRumEvent: any = {
+    id: 'img-id',
+    type: PERFORMANCE_RESOURCE_EVENT_TYPE,
+    details: {
+        fileType: ResourceType.IMAGE,
+        ...imagePerformanceEntry
+    }
+};
+
+const navigationRumEvent: any = {
+    id: 'nav-id',
+    type: PERFORMANCE_NAVIGATION_EVENT_TYPE,
+    details: navigationEvent
+};
+
+const mockLCPDataWithImage = Object.assign({}, mockLCPData, {
+    attribution: {
+        ...mockLCPData.attribution,
+        lcpResourceEntry: imagePerformanceEntry
+    }
+});
+
 jest.mock('web-vitals/attribution', () => {
     return {
-        onLCP: jest
-            .fn()
-            .mockImplementation((callback) => callback(mockLCPData)),
+        onLCP: jest.fn().mockImplementation((callback) => {
+            context.eventBus.dispatch(Topic.EVENT, imageResourceRumEvent);
+            context.eventBus.dispatch(Topic.EVENT, navigationRumEvent);
+            callback(mockLCPDataWithImage);
+        }),
```

Comment:
question: Should we test any integrations? I'd like some evidence that everything works together as intended.

E.g.,
1. Between nav/resource and web vitals plugin.
2. Between browser and web vitals plugin.
3. Between web vitals lib and web vitals plugin.

Reply:
Good idea. FWIW manual testing had no problems so far